Fix updating reports

ColinLeu opened the pull request:
If there were already some (successful) reports for the same commitId, uploading artifacts always failed, because the wrong amount of columns was returned by the inner SELECT statement.

The following error was returned by the server in that case: "{"description":"couldn't update flaky tests: sub-select returns 2 columns - expected 1","error":"Internal Server Error"}"

Therefore, a lot of flaky test runs were probably never actually counted.

ColinLeu committed:
Fix updating reports

If there were already some (successful) reports for the same commitId, uploading artifacts always failed,
because the wrong amount of columns was returned by the inner SELECT statement.

The following error was returned by the server in that case:
"{"description":"couldn't update flaky tests: sub-select returns 2 columns - expected 1","error":"Internal Server Error"}"

Therefore, a lot of flaky test runs were probably never actually counted.

UweBenary commented on database/test.go:
```diff
@@ -84,7 +84,7 @@ func UpdateFlakyTests(db *sqlx.DB, commitID int) error {
 	UPDATE tests SET flaky = true
 		WHERE tests.id in (
-			SELECT DISTINCT(tests.id), tests.name FROM results
+			SELECT DISTINCT(tests.id) FROM results
```
Eamonn has investigated. 

- Your solution fixes the 2 column issue. 
- FYI: This snippet should have always thrown. Since noflake has worked at least once there must have been something to prevent calling this UPDATE, ie there should be some other code to SET flaky = true. But we did not find it immediately. In other words there might be something else fishy. But this will be object to another issue then.
- The GROUP BY statement is obsolete as we DISTINCT

ColinLeu replied:
> The GROUP BY statement is obsolete as we DISTINCT

Isn't it the other way around? `DISTINCT(tests.id)` is redundant since we already GROUP BY (which is required to check if tests had successes AND failures) ?

> FYI: This snippet should have always thrown. Since noflake has worked at least once there must have been something to prevent calling this UPDATE, ie there should be some other code to SET flaky = true. But we did not find it immediately. In other words there might be something else fishy. But this will be object to another issue then.

Technically, it should have always thrown if there were tests which both failed AND succeeded for the same commitId. But why are flaky tests found if the update script does not work? 
Well, the flag `flaky` was added as a performance optimization (pre-calculating the flaky-ness). Tests which were uploaded before that change got the flag from a migration script (`3_flaky_tests.up.sql`). The query in that file is correct.

UweBenary replied:
agree

UweBenary commented:
File exists error seems to be ralated with lint and caching. Solution seams to be seat skip caching to true and/or move linting to later point in pipeline.

ColinLeu committed:
CI: Fix linting job

According to https://github.com/golangci/golangci-lint-action#how-to-use,
the linting job should be separated from other jobs like `go test`.

ColinLeu committed:
CI: Only trigger workflow on push

Otherwise, the same workflow is performed twice, when opening/updating a PR.

ColinLeu committed:
CI: Extract release step to dedicated workflow

This workflow should only be triggered on the main branch if the "Build" workflow succeeded.
